### PR TITLE
[device][ios] Added support for desktop detection

### PR DESCRIPTION
**Pull request by robertherber**
# Why

On iOS added support for deviceType detection of Desktop on MacOS, checking for Catalyst and iPad app running on Mac.

Supersedes #15084 which got outdated before merging, broken apart as requested in previous review.

# How

By checking the `ProcessInfo.processInfo.isMacCatalystApp` and `ProcessInfo.processInfo.isiOSAppOnMac` flags.

# Test Plan

Tested in the bare-expo app. And used in production through `patch-package`.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-device/ios/DeviceModule.swift [hunk 1]
```diff
@@ -47,6 +38,31 @@ public class DeviceModule: Module {
...
+  
+  // if it's built for iPad running on a Mac
+  if #available(iOS 14.0, *) {
+    if (ProcessInfo.processInfo.isiOSAppOnMac) {
+      return DeviceType.desktop.rawValue
+    }
+  }
+
+  switch UIDevice.current.userInterfaceIdiom {
+    case UIUserInterfaceIdiom.phone:
```
[the remark this answers is not in the document]
Comment (fobos531) [reply packages/expo-device/ios/DeviceModule.swift:55]: This can be fixed by running swiftlint against the `switftlint.yml` in this repository

**Review by fobos531 — CHANGES_REQUESTED** [review]

**Comment by robertherber** [comment]
@ide @tsapeta I'm all for linting - but getting linting feedback in CI is a recurring headscratcher when contributing to Expo packages. [Building SwifLint locally feels is definitely a threshold to contributing](https://github.com/realm/SwiftLint/blob/main/CONTRIBUTING.md)

**Comment by robertherber** [comment]
> > @ide @tsapeta I'm all for linting - but getting linting feedback in CI is a recurring headscratcher when contributing to Expo packages. [Building SwifLint locally feels is definitely a threshold to contributing](https://github.com/realm/SwiftLint/blob/main/CONTRIBUTING.md?rgh-link-date=2023-03-09T13%3A12%3A41Z) - is there a simpler way?
> 
> There's no need to build it locally. If you're on macOS (which I assume you are), there's a `swiftlint` homebrew formula which you can easily install. After that, position yourself in the `packages/expo-device` folder, run `swiftlint --fix` and that's all. Maybe this could be added to the contributing guide @tsapeta ?

I tried it now. The whole file intendation was changed and it still doesn't pass the CI 🤔 Please advice 😄

**Comment by robertherber** [comment]
@fobos531 @tsapeta @ide I've added #21654 with some notes about swiftlint in the contribution docs, I hope it can help other sporadic contributors as well, and keep the CI greener.. :)

**Comment by tsapeta** [comment]
@robertherber I'm sorry that the bot and linter get you annoyed. Right now the number of comments it can include in the review is limited to 10, because GitHub's API counts every comment as a notification (even though they all are sent in one http request) which is the basis of rate-limiting mechanism. We'll keep working on improving the contributors experience though. Thanks for submitting the PR to the contribution docs 🙏

**Review by tsapeta — APPROVED** [review]

**Pull request merged**